admin/osx/mac-crafter/Sources/Utils/Signer.swift

```diff
@@ -0,0 +1,187 @@
+            .appendingPathComponent("PlugIns")
+        
+        guard let enumerator = FileManager.default.enumerator(at: dynamicFrameworksLocation, includingPropertiesForKeys: nil) else {
+            fatalError("ERROR: Failed to get enumerator for: \(url.path)")
+        }
+        
+        let dynamicLibaries: [URL] = enumerator.compactMap { element in
+            guard let candidate = element as? URL else {
+                return nil
+            }
+            
+            guard candidate.path.contains(".appex/") == false else {
+                return nil
+            }
+            
+            guard candidate.pathExtension == "dylib" else {
+                return nil
+            }
+            
+            return candidate
+        }
+        
+        return dynamicLibaries
+    }
+    
+    ///
+    /// Find all extension bundles in the bundle at the given location.
+    ///
+    /// This assumes the internal structure of the bundle at the given location to have `Contents/PlugIns`.
+    ///
+    private static func findExtensions(at url: URL) throws -> [URL] {
+        let pluginsLocation = url
+            .appendingPathComponent("Contents")
+            .appendingPathComponent("PlugIns")
+        
+        var items = try FileManager.default.contentsOfDirectory(at: pluginsLocation, includingPropertiesForKeys: nil)
+        
+        items.removeAll { item in
+            if item.path.hasSuffix(".appex") == false {
+                return true
+            }
+            
+            print("Found extension bundle: \(item.path)")
+            return false
+        }
+        
+        return items
+    }
+
+    ///
+    /// Find all framework bundles in the bundle at the given location.
+    ///
+    /// This assumes the internal structure of the bundle at the given location to have `Contents/Frameworks`.
+    ///
+    private static func findFrameworks(at url: URL) throws -> [URL] {
+        let frameworksLocation = url
+            .appendingPathComponent("Contents")
+            .appendingPathComponent("Frameworks")
+        
+        var items = try FileManager.default.contentsOfDirectory(at: frameworksLocation, includingPropertiesForKeys: nil)
+        
+        items.removeAll { item in
+            if item.path.hasSuffix(".framework") == false {
+                return true
+            }
+            
+            print("Found framework bundle: \(item.path)")
+            return false
+        }
+        
+        return items
+    }
+    
+    private static func verify(at location: URL) async {
+        print("Verifying: \(location.path)")
+        await shell("codesign --verify --deep --strict --verbose=2 \"\(location.path)\"")
+    }
+
+    // MARK: - Public
+    
+    ///
+    /// Entry point for signing a whole desktop client app bundle.
+    ///
+    static func signMainBundle(at location: URL, codeSignIdentity: String, entitlements: [String: URL], developerBuild: Bool) async throws {
```

Comment:
The `developerBuild` argument is obsolete.

```suggestion
    static func signMainBundle(at location: URL, codeSignIdentity: String, entitlements: [String: URL]) async throws {
```